윤준석/윤준석.sql

```diff
@@ -0,0 +1,118 @@
+/* 
+과제 A:
+주어진 요구사항에 맞게 SQL 질의문을 만드세요
+강의 자료로 주어진 테이블을 통해 만드시면 됩니다.
+ */
+-- 1. 전공 이름이 '소프트웨어공학전공'인 남학생을 모두 조회하세요
+SELECT s.*
+FROM student s 
+JOIN major m ON s.major_id = m.major_id
+WHERE m.major_name = '소프트웨어공학전공' AND s.gender = 'male';
+
+-- 2. 30000원이 넘는 책의 이름과 가격을 조회하세요(가격에 대해 내림차순으로 정렬해야 합니다)
+SELECT b.title, b.price
+FROM book b
+WHERE b.price > 30000
+ORDER BY price DESC;
```

Comment:
질의문을 잘 작성해 주셨어요!
작성하신 것 처럼 사용해도 아무 문제 없지만, 참고하시면 좋을 것 같아서 댓글을 남깁니다.

`SELECT` 절에서 `b.title, b.price`처럼, 속성명 앞에 테이블명을 명시하고 있네요.
위처럼 `테이블명.속성명` 형식은, 주로 조인으로 인해 여러 테이블을 통해 SQL 질의문을 작성할 때 사용합니다.
여러 테이블을 한 질의문에 담다 보니 어떤 테이블에 속한 속성인지 명확하게 명시하려고 사용하기도 하고, 양 테이블에 똑같은 이름을 지닌 속성이 있으면 에러가 발생하기 때문에 `b.title`처럼 `book` 테이블의 `title`이라는 것을 지정해 주고자 사용하기도 합니다.

작성해주신 질의문을 살펴보면, `title`과 `price`가 어느 테이블에 속한 속성인지를 명확하게 하려고 `b.title, b.price`처럼 작성해 주신 것 같아요.
테이블을 한 개만 사용하는 질의문이니 아예 `title, price`처럼 적어도 될 것 같습니다.
혹은 `book` 테이블이라는것을 바로 보여주고 싶다면, `book`을 굳이 `b`라는 별명으로 지정하지 않고 `book.title, book.price`처럼 적어도 될 것 같습니다.

```suggestion
-- 2. 30000원이 넘는 책의 이름과 가격을 조회하세요(가격에 대해 내림차순으로 정렬해야 합니다)
SELECT title, price
FROM book
WHERE price > 30000
ORDER BY price DESC;

-- 혹은 이렇게 작성할 수 있습니다.
SELECT book.title, book.price
FROM book
WHERE price > 30000
ORDER BY price DESC;
```

Reply:
말씀해주신대로 b.을 사용하지 않는게 나은 것 같습니다!